testng-core/src/test/kotlin/test/hook/samples/BaseConfigurableSample.kt

```diff
@@ -0,0 +1,27 @@
+package test.hook.samples
+
+import java.lang.reflect.Method
+import org.testng.IConfigurable
+import org.testng.annotations.BeforeClass
+import org.testng.annotations.BeforeMethod
+import org.testng.annotations.BeforeSuite
+import org.testng.annotations.BeforeTest
+
+abstract class BaseConfigurableSample : IConfigurable {
```

Comment:
I think having a class mixing test cases and listener is a bad practice and except for very specific use cases I don't want to promote it in our tests.

Prefer to use `@Listeners` instead

Reply:
`IConfigurable` can be implemented by a test class as well and there's no need that it has to be available as a listener. So I dont think that rule of making it available ONLY via a `@Listeners` annotation would hold true and relevant. Also when it becomes a listener it becomes global, but when a class implements it, it is local. Making it as a `@Listeners` driven mechanism alters the intent of the test and I don't want to be doing that in a PR wherein I am only doing migration. So I am going to skip applying this change.